python/lsst/pipe/tasks/assembleCoadd.py

```diff
@@ -1362,3 +1374,327 @@ def detectClipBig(self, tempExpClipList, clipFootprints, clipIndices, maskClipVa
+            chiOneMap = self._snrToBinaryIm(chiIm)
+            epochCountImage += chiOneMap
+
+        self.log.debug("Generating Mask List. Second loop through warps")
+        spanSetArtifactList = []
+        spanSetNoDataMaskList = []
+
+        maxNumEpochs = max(1, self.config.temporalThreshold*len(tempExpRefList))
+        for warpRef, imageScaler in zip(tempExpRefList, imageScalerList):
+            mi = self._readAndComputeWarpDiff(warpRef, imageScaler, templateCoadd)
+            chiIm = self._makeChiIm(mi)
+            chiOneMap = self._snrToBinaryArr(chiIm.array)
+            outliers = afwImage.makeMaskFromArray(chiOneMap.astype(afwImage.MaskPixel))
+            outliers.setXY0(mi.getXY0())
+            spanSetList = afwGeom.SpanSet.fromMask(outliers).split()
+
+            # PSF-Matched warps have less available area (~the matching kernel) because the calexps
+            # undergo a second convolution. Pixels with data in the direct warp
+            # but not in the PSF-matched warp will not have their artifacts detected.
+            # NaNs from the PSF-matched warp therefore must be masked in the direct warp
+            nans = numpy.where(numpy.isnan(chiIm.array), 1, 0)
+            nansMask = afwImage.makeMaskFromArray(nans.astype(afwImage.MaskPixel))
+            nansMask.setXY0(chiIm.getXY0())
+            spanSetNoDataMask = afwGeom.SpanSet.fromMask(nansMask).split()
+
+            filteredSpanSetList = self._filterArtifacts(spanSetList, epochCountImage,
+                                                        maxNumEpochs=maxNumEpochs)
+            dilatedSpanSetList = [s.dilated(self.config.growMaskBy, afwGeom.Stencil.CIRCLE)
+                                  for s in filteredSpanSetList]
+            spanSetArtifactList.append(dilatedSpanSetList)
+            spanSetNoDataMaskList.append(spanSetNoDataMask)
+        return pipeBase.Struct(artifacts=spanSetArtifactList,
+                               noData=spanSetNoDataMaskList)
+
+    def computeAltMaskList(self, tempExpRefList, maskSpanSets):
+        """!
+        \brief Apply artifact span set lists to masks
+
+        @param tempExpRefList: List of data references to warps
+        @param imageScalerList: List of image scalers
+
+        return List of alternative masks
+
+        Add artifact span set list as "CLIPPED" plane and NaNs to existing "NO_DATA" plane
+        """
+        spanSetMaskList = maskSpanSets.artifacts
+        spanSetNoDataList = maskSpanSets.noData
+        altMaskList = []
+        for warpRef, artifacts, noData in zip(tempExpRefList, spanSetMaskList, spanSetNoDataList):
+            warp = warpRef.get(self.getTempExpDatasetName(self.config.warpType), immediate=True)
+            mask = warp.maskedImage.mask
+            maskClipValue = mask.addMaskPlane("CLIPPED")
+            noDataValue = mask.addMaskPlane("NO_DATA")
+            _ = [s.clippedTo(mask.getBBox()).setMask(mask, 2**maskClipValue) for s in artifacts]
```

Comment:
I think a plain `for` loop would make more sense here than assigning a list comprehension to a dummy variable. Same for the next line.